pkg/util/logger_windows.go

```diff
@@ -0,0 +1,134 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"strings"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/go-kit/kit/log"
+	"github.com/weaveworks/common/server"
+	el "golang.org/x/sys/windows/svc/eventlog"
+)
+
+// Default name for the Grafana Agent under Windows
+const ServiceName = "Grafana Agent"
+
+//NewWindowsEventLogger creates a new logger that writes to the event log
+func NewWindowsEventLogger(cfg *server.Config) *Logger {
+	l := Logger{makeLogger: makeWindowsEventLogger}
+	if err := l.ApplyConfig(cfg); err != nil {
+		panic(err)
+	}
+	return &l
+}
+
+func makeWindowsEventLogger(cfg *server.Config) (log.Logger, error) {
+	// error is necessary for the Log function to return an error
+	notAllowedError := errors.New("not_allowed")
+
+	// Setup the log in windows events
+	err := el.InstallAsEventCreate(ServiceName, el.Error|el.Info|el.Warning)
+
+	// We expect an error of 'already exists' for subsequent runs,
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return nil, err
+	}
+	il, err := el.Open(ServiceName)
```

Comment:
Is it valid to have two of these open at the same time? Can both of them still write?

With using the finalizer, it means there will be a window where two event logs for the same service name are used.

Reply:
Yes, and yes, I had double-checked that you could write to two but also tested it in parallel with no issue.